crates/common/tedge_config/src/tedge_toml/models/path.rs

```diff
@@ -20,11 +20,18 @@ impl AbsolutePath {
     }
 
     pub fn from_path(path: Utf8PathBuf) -> Result<Self, InvalidAbsolutePath> {
-        if path.is_relative() {
-            Err(InvalidAbsolutePath(path))
+        let absolute_path = if path.is_absolute() {
+            path
         } else {
-            Ok(AbsolutePath(path))
-        }
+            let Ok(Ok(pwd)) = std::env::current_dir().map(Utf8PathBuf::try_from) else {
+                return Err(InvalidAbsolutePath(path));
```

Comment:
Minor:
```suggestion
                return Err(InvalidRelativePath(path));
```
Else the error message will be slightly misleading, right? As the issue is not at all with the input `path` but from the derived `pwd`.

Reply:
I changed the error message to "Cannot be converted to an absolute path" but let the error type name unchanged because it makes sense for `AbsolutePath::try_new` to fail with `InvalidAbsolutePath`.


See https://github.com/thin-edge/thin-edge.io/pull/3564/commits/4134c9e92038199220a156b659fe208ad546fb4a